core/dbt/parser/base.py

```diff
@@ -206,11 +209,25 @@ def _update_parsed_node_info(self, parsed_node, config):
         # definition, not the current package
         schema_override = config.config.get('schema')
         get_schema = self.get_schema_func()
-        parsed_node.schema = get_schema(schema_override).strip()
+        try:
+            schema = get_schema(schema_override, parsed_node)
+        except dbt.exceptions.CompilationException as exc:
+            too_many_args = (
+                "macro 'dbt_macro__generate_schema_name' takes not more than "
+                "1 argument(s)"
+            )
+            if too_many_args not in str(exc):
+                raise
+            msg = ('The generate_schema_name macro does not accept a second '
```

Comment:
Can you change this to

```
As of dbt v0.14.0, the `generate_schema_name` macro accepts a second "node" argument.
The one-argument form of `generate_schema_name` is deprecated, and will become
unsupported in a future release.

For more information, see: https://docs.getdbt.com/v0.14/docs/upgrading-to-014
```

What's the benefit of rolling this as a warning (with `repeat=False`) over using a new Deprecation class?

Reply:
Honestly, I had completely forgotten about the existence of deprecations somehow. I guess I haven't had to think about them in a while...